### PR TITLE
Make nav drop-down cleaner/scrollable

### LINKED ISSUE #9768: mobile website: Menu can't be scrolled down (can't acccess User Icon, sync button, etc)
When viewing the website on a phone's browser, unless your phone screen is long, some of the menu entries can't be accessed. They don't appear on the screen and the menu can't be scrolled down to get to the hidden ones.

Here's a screenshot from Birkwelch (310aea01-13f5-4fb6-929e-be39d63e665b) who says "I have a Samsung J7 Galaxy Perx, and the issue occurs both in the Samsung internet browser and in Chrome. If you open the menu, I believe the user icon would show at the bottom, but you can't scroll down to see or reach it... instead, the tasks behind the menu scroll instead."

Screenshot with menu open

Here's what it's like on my phone:
screenshot_20180106-082947

Related to that, as a suggestion, the Gems, Gold, and Hourglasses could be hidden in the drop-down menu when using a mobile browser. It's kind of weird to see information like that in a menu. Perhaps on the mobile website they could be moved to the part of the header that's visible before you tap on the menu icon. On some phones, that would allow all menu items to be accessed without the need to scroll, so it would be a usability improvement for some people.

### PR DESCRIPTION
**Pull request by probaton**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #9768

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
- Made nav menu scrollable.
- Moved user menu to the top of the menu, as it seemed more important than the rest of the nav contents. 
- Reduced user menu to a single line, as the complete row should easily fit on even the smallest of phones. 
image

I tested the fix in Chrome and Firefox on both desktop and smartphone. I don't have suitable iOS/Mac devices, so I couldn't test those specifically. I have no reason to believe the fix won't work in Safari, but it might be prudent to check it out anyway. 

Note that opening the user/notification menu within the nav menu is still ugly as %*(#. I figured fixing that was outside the scope of the story, but if someone thinks it's terribly important (and can tell me what the desired result is) I suppose I could dive back into CSS hell. 

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: 6d7d6651-604d-4e7c-adff-a040770a6768

### CODE REVIEW

**Comment by probaton** [comment]
The Travis build is failing on a failed API integration test. Now I'm a test engineer, so I'm generally a skeptical human being, but even I have trouble believing that I knocked over a web service API with CSS changes. 

Presumably this is just general test flakiness? Is there some way I can kick off a new build? Or maybe people could give me pointers on how to make the test less flaky?

**Comment by paglias** [comment]
Some tests randomly fail, don't worry about them :) We usually restart them manually but in this case since it's just css changes it's not necessary.

I'm going to review every PR today, sorry for the delay but the team was travelling this and last week

**Comment by probaton** [comment]
Good to hear and not a problem (in that order).

**Comment by paglias** [comment]
@Tressley is this okay?

**Comment by Tressley** [comment]
@paglias / @probaton -- I think that this is okay, but I do have completed designs for the mobile navigation that could probably be lumped into this issue. Here's a quick screenshot for updated positioning.

image

Are you a member of our Zeplin project @probaton? If you haven't been invited, just drop an email here and I'll fire an invite your way.

**Comment by probaton** [comment]
@Tressley Got the Zeplin invite! That's definitely a lot more efficient than batting screenies back and forth. 

I'll leave it to you to decide if you want me to implement the entire design or accept my fix as is. On the one hand it does technically fix the issue. On the other hand I get that doing a weird interim fix doesn't make sense if we're overhauling the menu anyway. 

I hope to have a couple hours to spare later in the week; let me know if I'll be spending it finicking away in CSS or picking up something new.

**Comment by Tressley** [comment]
@probaton -- Great! I think it makes more sense to have you implement the approved designs as it will prevent us from just opening a new GitHub issue to overwrite your changes. What do you think? Still interested in seeing this one through? You can find the art boards here:

Mobile: https://zpl.io/VDDgMj3
Tablet: https://zpl.io/bJNDP69

Feel free to leave any questions you may have on the art boards!

**Comment by probaton** [comment]
@Tressley I was afraid you were going to say that. Be warned that, whatever impression I may inadvertently have given, I have virtually zero experience doing design work. So this is a) going to take a while and b) may explode in my face. 

On the other hand I did start contributing to expand my comfort zone and the deep end does look fancy, so sign me up!

**Comment by paglias** [comment]
In that case I'll remove the ready for review status from the PR @probaton , let me know if you need any help implementing the design

**Comment by lemoness** [comment]
@probaton Hi there! How is this fix going?

**Comment by probaton** [comment]
Slow. I would love to add 'steady', but I feel that would be giving me more credit than I deserve. I should have another hour to poke at it this afternoon.

**Comment by probaton** [comment]
Things are moving in the right direction, but I find myself putting a lot of effort in salvaging the on-hover drop-down behavior even though I agree with bootstraps default of only opening on click. Personally, I dislike on-hover effects in general, especially when the drop-down obscures other content. I often end up clicking the menu item before I realize the drop-down is automatic, thereby navigating to the wrong page and having to try again. 

Added bonus is that the code is cleaner (less customized behavior) and Habitica conforms to generally accepted interwebz standards. 

Is everyone really invested in the hover drop-downs or might I convince you to revert to bootstrap's on-click behavior?

**Comment by Tressley** [comment]
@probaton -- Perhaps I can help! For desktop, hovering menu items to expand is not uncommon. For tablets and mobile, hover isn't exactly an option, so having opening on click would make sense for the smaller breakpoints. Also, the new menu can push the content down instead of overlaying it, if you'd like to give that a try.

**Comment by probaton** [comment]
@Tressley Thanks for the offer, but I'm doing ok for the moment. I do have a couple ideas on how to fix it, that's not necessarily the problem. It's that I'm genuinely not a big fan of the on-hover menus, an opinion apparently shared by Bootstrap. As such I wasn't particularly invested in implementing a relatively complicated workaround for something I wouldn't consider broken in the first place. 

Obviously it's your call, if you guys are set on on-hover menus then that's what I'll build, I just wanted to make the case for the alternative. 

Good point on the on-click for mobile, as well as the push contents instead of overlay. I'd already come to the same conclusions, so both features are already in the new version. I can do a push of my preliminary tweaks if you want to take a look.

**Comment by beffymaroo** [comment]
@Tressley ( @probaton ) What's the status of this fix? Checking in since it's marked Important and we do have a fair amount of user comments/requests to get this working again.

**Comment by probaton** [comment]
Holidays, a busy schedule, and my own complete lack of talent have not been kind, but it's actually starting to look pretty good! I've mostly finished restructuring the header to accommodate the new changes, so I'm mainly adding new colors and borders and such to match the new design.

Again, if there's an acute need for a fix the branch as it's been pushed will actually solve the main issue. I can always submit a new PR for the design update later.

**Comment by probaton** [comment]
So I actually think I may be done. Ish. Everything looks the way it should as far as I can tell, though I am, as ever, eminently amenable to comments. 

Full disclosure; everything was looking pretty and peachy yesterday, but then I rebased to fix a few merge conflicts and since then I haven't been able to get my clone running (on either my branch or upstream/develop). As far as I know the conflicts were all pretty innocent, so I'm reasonably sure everything should be fine. However, if you'd prefer I did a final check myself you're going to have to wait until I've figured out why my set-up is borked. 

As a side note I spent a _lot_ of time trying to figure out what the preferred code styles were. For starters, I spent a lot of time removing all the bootstrap overrides (as my mother always taught me that's a bad habit to get into) but there were so many that maybe we have a reason to want to do that? And then I found three (four?) different methods of adding bootstrap classes/styles, all mixed together. All in all I think I spent as much time trying to figure out if the behavior I was 'fixing' was default or custom as I did actually solving problems. 

TL;DR: I think a coding style guide of some sort might have saved me a bunch of trouble. To paraphrase Douglas Adams; that should help in getting things right, or at least definitively wrong.

**Comment by paglias** [comment]
@probaton thanks for the update but I think there are some problems, this is chrome with the latest changes

schermata 2018-05-17 alle 12 01 27


Let me know if you need help with your local setup

**Comment by probaton** [comment]
Was that not the desired result? I thought it was a huge improvement!

On a more serious note: arg. Looks like my conflict resolution borked something after all, which means I'm going to have to ask you to be patient until I can get my local install working again. 

I've already posted a description in the aspiring blacksmiths guild, but in case someone here could spare a moment to help me out I'll leave the link to my bug report here:
https://docs.google.com/document/d/1XZceXUOk94PTv0qFShY1WMye3nrKUbpvMubAq7vKpDw/edit

The abridged version is that my terminal starts spamming an error every few seconds every time I try `npm start`. 

```
2018-05-16T08:35:28.227Z - info: [Wed May 16 2018 10:35:28 GMT+0200 (CEST)] [master:6735] worker:6941 disconnect! new worker:6947 fork
/home/snickersnacker/dev/habitica/website/server/models/task.js:166
TaskSchema.statics.findByIdOrAlias = async function findByIdOrAlias(identifier, userId, additionalQueries = {}) {
                                           ^^^^^^^^
SyntaxError: Unexpected token function
```

**Comment by paglias** [comment]
Which node version are you using? It should be node 8

Il Gio 17 Mag 2018, 13:02 Dexx Mandele <notifications@github.com> ha
scritto:

**Comment by probaton** [comment]
@paglias I was totally convinced I was using node 8 until I finally figured out `nvm use` only applies to the current terminal session. Arg. 

Thanks for pointing me in the right direction! I'll get started on the finishing touches.

**Comment by probaton** [comment]
Predictably, it was a missing curly brace, sorry. Being an idiot is exhausting. 

Things are once again looking shiny; I'm looking forward to your feedback.

**Comment by paglias** [comment]
Thanks! That looks great, we'll merge it tomorrow and then test for a few days before it goes online

**Comment by paglias** [comment]
Merged @probaton ! I've notated this PR towards your 3rd tier :)
